bcgov/House-Innovations-HOUSNAV
# HOUSNAV-22: question footer and initial walkthrough state stores

[HOUSNAV-22](https://hous-bssb.atlassian.net/browse/HOUSNAV-22)

## Overview & Purpose
These changes add a footer that looks like the design and improves the overall walkthrough display to be close to the design.

## Summary of Changes
Added a WalkthroughStore
Updated footer, walkthrough, and question components to better match the design
Updated walkthrough and child components to consume the WalkthroughStore
Added the tertiary button type for the back button in the footer
Added icons for the buttons in the footer

## Side Effects
The includes a complete refactoring of how the walkthrough was handling state. It now happens in a central store. This store is added to a react Provider at the top level of the walkthrough. Each component in the walkthrough can now observe the store and react to changes as needed on it's own.

## Testing
1. Navigate to the walkthrough
2. Observe the disabled buttons in the footer to match the design
3. Click an answer and observe the next button being enabled
4. Click next and see question #2, observe the next button is disabled again
5. Observe the back button is now enabled.
6. Click back button and observe question #1 again with previous answer shown.
7. Click back and forth observing state change of footer and questions
8. Observe question scrolling back to the top if next/back is clicked while scrolled father down the question

## Notes & Resources
There are only 3 possible questions. Click next on question #3 just goes back to question #2.
